packages/jest-resolve/src/resolver.ts

```diff
@@ -461,6 +461,10 @@ export default class Resolver {
     );
   }
 
+  isNodePath(moduleName: string): boolean {
+    return moduleName.startsWith('node:');
+  }
```

Comment:
Just above `isCoreModule` method has the same check: `moduleName.startsWith('node:'))`. The logic of removing `'node:` prefix is also repeated here and there. Wouldn’t it be cleaner to normalize the module specifier before passing it around with something like this:

```ts
normalizeModuleSpecifier(specifier: string): string {
  return specifier.startsWith('node:')
    ? specifier.slice('node:'.length)
    : specifier;
}
```